TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/Robot.java

```diff
@@ -2,73 +2,150 @@
 
 import androidx.annotation.NonNull;
 
+import com.qualcomm.hardware.bosch.BNO055IMU;
 import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.hardware.ColorSensor;
 import com.qualcomm.robotcore.hardware.DcMotor;
-import com.qualcomm.robotcore.hardware.DcMotorSimple;
+import com.qualcomm.robotcore.hardware.DcMotorEx;
+import com.qualcomm.robotcore.hardware.NormalizedColorSensor;
+import com.qualcomm.robotcore.hardware.Servo;
 
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
 import org.firstinspires.ftc.teamcode.api.bp.AbstractRobot;
-import org.firstinspires.ftc.teamcode.api.vars.Hardware;
+import org.firstinspires.ftc.teamcode.api.bp.ArmRobot;
+import org.firstinspires.ftc.teamcode.api.bp.WheeledRobot;
+import org.firstinspires.ftc.teamcode.api.config.Constants;
+import org.firstinspires.ftc.teamcode.api.config.Naming;
+import org.firstinspires.ftc.teamcode.api.hw.Gyroscope;
+import org.firstinspires.ftc.teamcode.api.hw.SmartColorSensor;
+import org.firstinspires.ftc.teamcode.api.hw.SmartMotor;
+import org.firstinspires.ftc.teamcode.api.hw.SmartServo;
 
-public class Robot extends AbstractRobot {
-    public DcMotor frontLeft, frontRight, backLeft, backRight, duck;
-
-    public Robot() {
-        super();
-    }
+public class Robot extends AbstractRobot implements WheeledRobot, ArmRobot {
```

Comment:
Most of the functions here are redundant (implemented in the movement class). Pick which ones you would want to keep